mindsdb/integrations/handlers/dremio_handler/dremio_handler.py

```diff
@@ -0,0 +1,236 @@
+from typing import Optional
+from collections import OrderedDict
+
+import json
+import time
+import requests
+import pandas as pd
+
+from mindsdb_sql import parse_sql
+from mindsdb_sql.render.sqlalchemy_render import SqlalchemyRender
+from mindsdb.integrations.libs.base import DatabaseHandler
+from sqlalchemy_dremio.base import DremioDialect
+
+from mindsdb_sql.parser.ast.base import ASTNode
+
+from mindsdb.utilities import log
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+
+class DremioHandler(DatabaseHandler):
+    """
+    This handler handles connection and execution of the Dremio statements.
+    """
+
+    name = 'dremio'
+
+    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+        """
+        Initialize the handler.
+        Args:
+            name (str): name of particular handler instance
+            connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'dremio'
+
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.base_url = f"http://{self.connection_data['host']}:{self.connection_data['port']}"
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> dict:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        headers = {
+            'Content-Type': 'application/json',
+        }
+
+        data = '{' + f'"userName": "{self.connection_data["username"]}","password": "{self.connection_data["password"]}"' + '}'
+
+        response = requests.post(self.base_url + '/apiv2/login', headers=headers, data=data, verify=False)
```

Comment:
For other calls we used apiv3, is it required to use v2 for login?

Reply:
Yes, it looks like the authentication still uses an older version of the API,
https://docs.dremio.com/software/rest-api/#authentication